binary-quantization-mistral/bq-mistral-embed.md

```diff
@@ -0,0 +1,537 @@
+
+The output confirms the initial condition—the collection was indeed empty and the upsert operation began. The logged information and the completion of the progress bar signify not only the start but also the successful end of the operation, with all batches of points inserted into the collection. This step is fundamental for subsequent operations, such as querying or searching the collection, as it ensures that there's data within the collection to act upon.
+
+```python
+collection_info = client.get_collection(collection_name=collection_name)
+
+if collection_info.vectors_count == 0:
+    logger.info("Collection is empty. Begin upsert.")
+    bs = 1000  # Batch size
+    for i in tqdm(range(0, len(points), bs)):
+        slice_points = points[i : i + bs]  # Create a slice of bs points
+        client.upsert(collection_name=collection_name, points=slice_points)
+    
+```
+### Understanding Collection Information with Qdrant
+
+When working with Qdrant, a powerful vector search engine for managing large volumes of data, it's essential to grasp how to interact with collections. Collections in Qdrant are akin to tables in traditional databases and are where your data lives. To begin understanding the state and scale of your data within these collections, you can retrieve information about them through the Qdrant client. 
+
+One crucial piece of information you might be interested in is the count of vectors within a collection. Vectors are essentially the data entries that you've stored in a collection, often representing complex data types like images or text in a high-dimensional space, suitable for similarity search tasks.
+
+By executing a Python command like `client.get_collection(collection_name=collection_name).vectors_count`, you are invoking a method to fetch details about a specific collection, identified by `collection_name`. Among the details returned, `vectors_count` is of particular interest as it reveals the total number of vectors stored in the collection.
+
+The output of `['']` suggests that either the code snippet did not execute as expected, or the collection currently contains zero vectors. This insight would typically prompt further investigation to ensure that data ingestion processes are functioning correctly or to validate that the specified collection is the intended target for interaction. Understanding and monitoring the vectors count is critical for both operational management and performance optimization within Qdrant, making it an essential tool in the arsenal of any data engineer or scientist working with this technology.
+
+```python
+collection_info = client.get_collection(collection_name=collection_name)
+collection_info.vectors_count
+```
```

Comment:
`vectors_count` and other `counts` are not reliable (they are approximate, e.g. AFAIK, they do not take into account vector deletions), to check the number of points in the collection one should use `count API`